napari/utils/progress.py

```diff
@@ -0,0 +1,138 @@
+import inspect
+from typing import Iterable, Optional
+
+from tqdm import tqdm
+
+_tqdm_kwargs = {
+    p.name
+    for p in inspect.signature(tqdm.__init__).parameters.values()
+    if p.kind is not inspect.Parameter.VAR_KEYWORD and p.name != "self"
+}
+
+
+class progress(tqdm):
```

Comment:
Is there a reason this class is `progress` and not `Progress`?

Reply:
Honestly, no. I went with lowercase because it's used mainly as a wrapper and it somehow feels nicer to write 
`for _ in progress(range(...))` than `for _ in Progress(range(...))`. I'm not married to either though :shrug:

Reply:
Maybe @tlambert03 or @jni have thoughts on what the appropriate convention is here. I don't mind either way too, maybe I also have a slight preference for lowercase given the usage ...

Reply:
Hmm, I thought I left a comment as part of my review... 🤷 Anyway, yes, it's common in Python to make classes lowercase when they are intended to be used as a function/context manager — case in point, tqdm itself. So I'm happy with the lowercase here.